bin2tif/terra_bin2tif.py

```diff
@@ -80,81 +83,169 @@ def process_message(self, connector, host, secret_key, resource, parameters):
+                shutil.move(out_tmp_tiff_left, left_tiff)
+                if left_tiff not in resource['local_paths']:
+                    fileid = upload_to_dataset(connector, host, self.clowder_user, self.clowder_pass, target_dsid,
+                                               left_tiff)
+                    uploaded_file_ids.append(host + ("" if host.endswith("/") else "/") + "files/" + fileid)
+                else:
+                    self.log_info(resource, "file found in dataset already; not re-uploading")
+                self.created += 1
+                self.bytes += os.path.getsize(left_tiff)
             else:
-                self.log_info(resource, "file found in dataset already; not re-uploading")
-            self.created += 1
-            self.bytes += os.path.getsize(left_tiff)
+                self.log_info("Zero bytes file generated")
 
         if (not file_exists(right_tiff)) or self.overwrite:
             self.log_info(resource, "creating & uploading %s" % right_tiff)
-            right_image = terraref.stereo_rgb.process_raw(right_shape, img_right, None)
-
-            create_geotiff(right_image, right_gps_bounds, out_tmp_tiff, None, False, self.extractor_info, metadata)
-            shutil.move(out_tmp_tiff, right_tiff)
-            if right_tiff not in resource['local_paths']:
-                fileid = upload_to_dataset(connector, host, self.clowder_user, self.clowder_pass, target_dsid,right_tiff)
-                uploaded_file_ids.append(host + ("" if host.endswith("/") else "/") + "files/" + fileid)
+            right_image = terraref.stereo_rgb.process_raw(left_shape, img_right, None)
+            out_tmp_tiff_right = os.path.join(tempfile.gettempdir(), resource['id'].encode('utf8'))
+            # Rename output.tif after creation to avoid long path errors
+            create_geotiff(right_image, gps_bounds, out_tmp_tiff_right, None, True, self.extractor_info, terra_md_full)
+            if os.path.getsize(out_tmp_tiff_right) > 0:
+                shutil.move(out_tmp_tiff_right, right_tiff)
+                if left_tiff not in resource['local_paths']:
+                    fileid = upload_to_dataset(connector, host, self.clowder_user, self.clowder_pass, target_dsid,
+                                               left_tiff)
+                    uploaded_file_ids.append(host + ("" if host.endswith("/") else "/") + "files/" + fileid)
+                else:
+                    self.log_info(resource, "file found in dataset already; not re-uploading")
+                self.created += 1
+                self.bytes += os.path.getsize(left_tiff)
             else:
-                self.log_info(resource, "file found in dataset already; not re-uploading")
-            self.created += 1
-            self.bytes += os.path.getsize(right_tiff)
+                self.log_info("Zero bytes file generated")
+
+
+        # Plot dir is the day under Level_1_Plots/ir_geotiff/day
+        # TODO
```

Comment:
this whole block should be the new plot clipping extractor we need to write. we will submit the files we've just uploaded to clowder to it during THIS extractor, and then that extractor will do this clipping bit

Reply:
fixed